server/service/src/asset/insert.rs

```diff
@@ -28,6 +30,7 @@ pub struct InsertAsset {
     pub catalogue_item_id: Option<String>,
     pub installation_date: Option<NaiveDate>,
     pub replacement_date: Option<NaiveDate>,
+    pub location_ids: Option<Vec<String>>,
```

Comment:
If it doesn't cause a problem for the front end, I'd probably remove this from the `InsertAsset` struct.
We don't all people to assign locations during the creation of a new asset, so we only really need this extra logic on update. That way it's also only in one place which I like.